File: first-apis/home/models/books.py
from datetime import date
import datetime
from django.db import models
from home.models.publishers import Publishers
from home.models.authors import Authors



class Books(models.Model):
    
 
    id   = models.AutoField(primary_key=True)
    name = models.CharField(max_length=225 ,null=True,blank=True,default=False)
    content = models.CharField(max_length=225 ,null=True,blank=True,default=False)
    page = models.CharField(max_length=225 ,null=True,blank=True,default=False)
    created_at= models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)
    
   # publishers = models.OneToOneField(Publishers,on_delete=models.CASCADE, null=True)
   # authors = models.OneToOneField(Authors,on_delete=models.CASCADE, null=True)
   # title = models.CharField(max_length=100 , null=True,blank=True,default=False)
    
    authors = models.ForeignKey(Authors, on_delete=models.CASCADE, related_name='authors', null=True)
    
    #authors = models.ManyToManyField(Authors)
    
    class Meta:
        db_table = 'books'
        

        
    
    
    

File: newapis/src/app/config/books/books.ts
export class Books {
    id?: any;
    content?: string;
    name?: string;
    page?: string;
    authors_id?: number;
  }

  export class BooksC {
    content?: string;
    name?: string;
    page?: string;
    authors_id?: number;
    constructor(obj?: Books) {
        this.name = obj && obj.name || ''
        this.content = obj && obj.content || ''
        this.page = obj && obj.page || ''
        this.authors_id = obj && obj.authors_id || 0
    }
}